### PR TITLE
Feature/737 isreal iscomplex

### PR DESCRIPTION
**Pull request by mtar**
## Description

Add iscomplex() and isreal() functionality that checks element-wise for complex or real values.

Issue/s resolved: #737 

## Changes proposed:
- New feature iscomplex
- New feature isreal

## Type of change

- New feature (non-breaking change which adds functionality)

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/types.py [hunk 1]
```diff
@@ -607,6 +611,42 @@ def can_cast(from_, to, casting="intuitive"):
...
+    --------
+    >>> ht.iscomplex(ht.array([1+1j, 1]))
+    DNDarray([ True, False], dtype=ht.bool, device=cpu:0, split=None)
+    """
+    sanitation.sanitize_in(x)
+
+    if issubclass(x.dtype, _complexfloating):
+        return x.imag != 0
+    else:
+        return factories.zeros(x.shape, bool, split=x.split, device=x.device, comm=x.comm)
```
Comment (coquelin77) [inline heat/core/types.py:632]: why not use `torch.is_complex`?

**Review by coquelin77 — CHANGES_REQUESTED** [review]

File: heat/core/types.py (hunk 1 above)
Comment (mtar) [reply heat/core/types.py:632]: I did not find it :mag: :smile:

Comment (mtar) [reply heat/core/types.py:632]: torch.is_complex returns the result for the whole tensor, not element-wise

**Comment by mtar** [comment]
run tests

File: heat/core/types.py (hunk 1 above)
Comment (coquelin77) [reply heat/core/types.py:632]: wow. that is a really annoying difference. but that is what it is

**Review by coquelin77 — APPROVED** [review]

**Pull request merged**